### ISSUE TITLE
DoExpansionWarningStatement

### ISSUE THREAD

**Issue by ilteroi** [LoneGazebo/Community-Patch-DLL#8338]
you sometimes get an expansion warning from AI players after founding a city far away from them.

it seems to happen because DoExpansionWarningStatement has a debouncing timer which depends on the last settled city, no matter where that city is.

is that by design?

**Comment by RecursiveVision** [comment]
@ilteroi When you respond to the first warning it'll set "expansion promise state" to "made" or "ignored". So if you choose to ignore it you'll get a second warning as you describe, but if you make it, it'll set "ever made expansion promise" to true and no further warnings will be received.

Either way it's a flawed system...

**Comment by RecursiveVision** [comment]
fixed

**Comment by RecursiveVision** [comment]
@ilteroi I took out all the "closest city pair" code (moving that to CvPlayer since it's used for the proximity calculation) and replaced it with a flat 6-8 tile "pissy range" value based on Boldness and the leader's UA. Super simple and easy.

Additionally I added a memory value to CvCity; once a city has triggered bickering and either a warning is sent or a promise is flagged as broken, all the triggering player's cities will be marked as ignored for future bickering. This prevents exploits wherein the new city had to be closer than the previous one to trigger the AI getting mad.
